### PR TITLE
fix(agw): Modified code to handle TAU Req for connected UEs

### LINKED ISSUE #11050: bug[MME] unknown-mme-ue-s1ap-id In a specific multi-sim UE scenario
Your Environment:

- Version: 1.7.0-1640913705-e6e795fd (master)
- Affected Component: AGW MME
- Deployment Environment: bare-metal (AGW)

Hi, I'm testing a scenario when Magma acts as a core for data only mobile LTE network.

Environment: **Dual Sim UE**.
One sim slot is for data only services (our LTE network with Magma in a center).
Second sim slot is for voice and sms service from regular mobile operator.

I found inconvenience with some dual sim dual standby UE like Samsung A01s, Redmi Note 9 and other:
When multiple sms come to mobile operator sim, our sim card inserted to first slot lost connection for few minutes.
Many different combination of Android configuration has been tested, like sim card priority, network type restrictions, etc.
No luck.

To Reproduce:
1. Some UE like Samsung A01s, Redmi Note 9 with two sim card activated. One sim with data services, another one with voice and sms.
2. Send multiple sms in a short period of time to sim with voice and sms services. This behaivior is common for many  sms chat application.

Here is log and capture:
[magmaDoubleSimTest.zip](https://github.com/magma/magma/files/7809271/magmaDoubleSimTest.zip)

### PR DESCRIPTION
**Pull request by rsarwad**
fix(agw): Modified code to handle TAU Req for connected UEs
## Summary
1. Added code to handle TAU req received in initial ue message for connected UEs.
2. Since UE is connected, mme shall release the previous s1 signaling connection. Added ue_context release timer at s1ap to track the ue context release complete message from eNB.
3. Release the UE contexts either on receiving ue context release complete message or on expiry of ue_context release timer.
4. While handling handover scenarios; only s1 signaling needs to be removed and shall not release the UE contexts.
## Test Plan
Executed s1ap sanity test suite, for which test_multi_enb_complete_reset.py was failing because as part of this TC; it first connects 5 eNBs and for each eNB there shall be one UE connected.
When complete reset is initiated for one eNB; but all UE contexts were getting deleted; Ideally we need to delete the contexts only for the UE for which eNB Reset is triggered. The changes are available in PR, https://github.com/facebookexperimental/S1APTester/pull/70

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c [hunk 1]
```diff
@@ -1344,6 +1344,14 @@ status_code_e s1ap_mme_generate_ue_context_release_command(
   free(buffer);
   rc = s1ap_mme_itti_send_sctp_request(&b, assoc_id, stream, mme_ue_s1ap_id);
 
+  // For handover; intention is release the s1-signaling connection with
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c]: // For handover, release the s1-signaling connection with

File: lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c [hunk 2]
```diff
@@ -1344,6 +1344,14 @@ status_code_e s1ap_mme_generate_ue_context_release_command(
...
+  //  source eNB and retain the UE contexts mme_app, nas and spgw. Since
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c]: // source eNB and retain the UE contexts at mme_app, nas and spgw. Since

**Review by ssanadhya — APPROVED** [review]
lgtm

File: lte/gateway/python/magma/enodebd/BUILD.bazel [hunk 3]
```diff
@@ -32,10 +32,10 @@ py_binary(
     visibility = ["//visibility:public"],
     deps = [
         ":enodebd_lib",
+        "//lte/gateway/python/magma/enodebd/tr069:server",
         "//lte/protos:mconfigs_python_proto",
         "//orc8r/gateway/python/magma/common:sentry",
         "//orc8r/gateway/python/magma/common:service",
-        "//lte/gateway/python/magma/enodebd/tr069:server",
```
Comment (electronjoe) [inline lte/gateway/python/magma/enodebd/BUILD.bazel:38]: Thanks for fixing the ordering on these! Don't know how they slipped in, buildifier should have caught them on prior PRs.

**Review by themarwhal — APPROVED** [review]
Bazel changes lgtm!

**Pull request merged**